test-app/backend/src/shift/shift-api.ts

```diff
@@ -62,4 +66,65 @@ router.post('/new-shift', async (req, res) => {
   );
 });
 
+const errorCheckDates = (newShift: Partial<Shift>, origShift: Shift) => {
+  if (newShift.start && newShift.end && (newShift.start.getTime() > newShift.end.getTime())) {
+    return false;
+  } else if (newShift.start) {
+    if ((newShift.start.getTime() > origShift.end.getTime())) {
+      return false;
+    }
+    return true;
+  } else if (newShift.end) {
+    if (origShift.start.getTime() > newShift.end.getTime()) {
+      return false;
+    }
+    return true;
+  } else {
+    return true;
+  }
+};
+
+const partialShiftPreProcessor = (myShift: Partial<Shift>) => {
+  if (myShift.start) {
+    myShift.start = new Date(myShift.start);
+  }
+  if (myShift.end) {
+    myShift.end = new Date(myShift.end);
+  }
+  return myShift;
+};
+
+router.patch('/edit-shift/:id', async (req, res) => {
+  console.log('PATCH: Editting Calender Event...');
+  const id = req.params.id;
+  const newShiftParams = partialShiftPreProcessor(req.body as Partial<Shift>);
```

Comment:
Related to comment below about verifying the properties/keys, before you cast the `req.body`, you need to check that all keys are valid and part of Shift. This is sorta annoying. For now, you can just add a comment above this mentioning that.